src/Program.cs

```diff
@@ -26,21 +28,51 @@ static void Main(string[] args)
                 if (File.Exists(inputPath))
                 {
                     CommandLineUtils.CreateOutputDirectory(outputPath);
-                    HtmlProcessor.ProcessFile(inputPath, outputPath);
+
+                    // Check file extension to decide whether to use MdProcessor or HtmlProcessor
+                    string fileExtension = Path.GetExtension(inputPath);
+                    if (fileExtension == ".md")
+                    {
+                        MdProcessor.ProcessFile(inputPath, outputPath);
+                    }
+                    else if (fileExtension == ".txt")
+                    {
+                        HtmlProcessor.ProcessFile(inputPath, outputPath);
+                    }
+                    else
+                    {
+                        CommandLineUtils.Logger($"Unsupported file format for input path: {inputPath}");
+                    }
                 }
                 else if (Directory.Exists(inputPath))
                 {
-                    string[] files = Directory.GetFiles(inputPath, "*.txt");
+                    // Get all files in the directory and save them to the files array
+                    string[] files = Directory.GetFiles(inputPath, "*.*"); 
```

Comment:
Should not be allowing "*.*". The program should _only_ accept ".txt" or ".md". please revert this change and add an `or` condition for ".md"